<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Unfiltered human PBMCs (10X Genomics) â€¢ scRNAseqBasicWorkflow</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Unfiltered human PBMCs (10X Genomics)">
<meta property="og:description" content="scRNAseqBasicWorkflow">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scRNAseqBasicWorkflow</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/tenx-unfiltered-pbmc4k.html">Unfiltered human PBMCs (10X Genomics)</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/aedin/scRNAseqBasicWorkflow" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Unfiltered human PBMCs (10X Genomics)</h1>
                        <h4 data-toc-skip class="author">Aedin Culhane,
Michael Lynch, Maria Doyle</h4>
            
            <h4 data-toc-skip class="date">Dec 2nd, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/aedin/scRNAseqBasicWorkflow/blob/HEAD/vignettes/tenx-unfiltered-pbmc4k.Rmd" class="external-link"><code>vignettes/tenx-unfiltered-pbmc4k.Rmd</code></a></small>
      <div class="hidden name"><code>tenx-unfiltered-pbmc4k.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Here, we describe a brief analysis of the peripheral blood
mononuclear cell (PBMC) dataset from 10X Genomics <span class="citation">[@zheng2017massively]</span>. The data are publicly
available from the <a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/2.1.0/pbmc4k" class="external-link">10X
Genomics website</a>, from which we download the raw gene/barcode count
matrices, i.e., before cell calling from the <em>CellRanger</em>
pipeline.</p>
<p>This tutorial is edited and extracted from the OSCA Unfiltered human
PBMCs workflow <a href="https://github.com/OSCA-source/OSCA.workflows/blob/master/inst/book/tenx-unfiltered-pbmc4k.Rmd" class="external-link">tenx.unfiltered-pbmc4k.Rmd</a></p>
</div>
<div class="section level2">
<h2 id="data-loading">Data loading<a class="anchor" aria-label="anchor" href="#data-loading"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DropletTestFiles</span><span class="op">)</span></span>
<span><span class="va">raw.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DropletTestFiles/man/listTestFiles.html" class="external-link">getTestFile</a></span><span class="op">(</span><span class="st">"tenx-2.1.0-pbmc4k/1.0.0/raw.tar.gz"</span><span class="op">)</span></span>
<span><span class="va">out.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"pbmc4k"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="va">raw.path</span>, exdir<span class="op">=</span><span class="va">out.path</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DropletUtils</span><span class="op">)</span></span>
<span><span class="va">fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">out.path</span>, <span class="st">"raw_gene_bc_matrices/GRCh38"</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DropletUtils/man/read10xCounts.html" class="external-link">read10xCounts</a></span><span class="op">(</span><span class="va">fname</span>, col.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/uniquifyFeatureNames.html" class="external-link">uniquifyFeatureNames</a></span><span class="op">(</span></span>
<span>    <span class="fu">rowData</span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span><span class="op">$</span><span class="va">ID</span>, <span class="fu">rowData</span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span><span class="op">$</span><span class="va">Symbol</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v86</span><span class="op">)</span></span>
<span><span class="va">location</span> <span class="op">&lt;-</span> <span class="fu">mapIds</span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v86</span>, keys<span class="op">=</span><span class="fu">rowData</span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span><span class="op">$</span><span class="va">ID</span>, </span>
<span>    column<span class="op">=</span><span class="st">"SEQNAME"</span>, keytype<span class="op">=</span><span class="st">"GENEID"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quality-control">Quality control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h2>
<p>We perform cell detection using the <code><a href="https://rdrr.io/pkg/DropletUtils/man/emptyDrops.html" class="external-link">emptyDrops()</a></code>
algorithm, as discussed in the book <a href="https://bioconductor.org/books/3.16/OSCA.advanced/droplet-processing.html#testing-for-empty-droplets" class="external-link">Advanced
Single-Cell Analysis with Bioconductor</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">e.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DropletUtils/man/emptyDrops.html" class="external-link">emptyDrops</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="va">sce.pbmc</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">e.out</span><span class="op">$</span><span class="va">FDR</span> <span class="op">&lt;=</span> <span class="fl">0.001</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">unfiltered</span> <span class="op">&lt;-</span> <span class="va">sce.pbmc</span></span></code></pre></div>
<p>We use a relaxed QC strategy and only remove cells with large
mitochondrial proportions, using it as a proxy for cell damage. This
reduces the risk of removing cell types with low RNA content, especially
in a heterogeneous PBMC population with many different cell types.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/perCellQCMetrics.html" class="external-link">perCellQCMetrics</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, subsets<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Mito<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">location</span><span class="op">==</span><span class="st">"MT"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">high.mito</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/isOutlier.html" class="external-link">isOutlier</a></span><span class="op">(</span><span class="va">stats</span><span class="op">$</span><span class="va">subsets_Mito_percent</span>, type<span class="op">=</span><span class="st">"higher"</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="va">sce.pbmc</span><span class="op">[</span>,<span class="op">!</span><span class="va">high.mito</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">high.mito</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    Mode   FALSE    TRUE </span></span>
<span><span class="co">## logical    3985     315</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colData</span><span class="op">(</span><span class="va">unfiltered</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">unfiltered</span><span class="op">)</span>, <span class="va">stats</span><span class="op">)</span></span>
<span><span class="va">unfiltered</span><span class="op">$</span><span class="va">discard</span> <span class="op">&lt;-</span> <span class="va">high.mito</span></span>
<span></span>
<span><span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">unfiltered</span>, y<span class="op">=</span><span class="st">"sum"</span>, colour_by<span class="op">=</span><span class="st">"discard"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Total count"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">unfiltered</span>, y<span class="op">=</span><span class="st">"detected"</span>, colour_by<span class="op">=</span><span class="st">"discard"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Detected features"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">unfiltered</span>, y<span class="op">=</span><span class="st">"subsets_Mito_percent"</span>,</span>
<span>        colour_by<span class="op">=</span><span class="st">"discard"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Mito percent"</span><span class="op">)</span>,</span>
<span>    ncol<span class="op">=</span><span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="tenx-unfiltered-pbmc4k_files/figure-html/unref-unfiltered-pbmc-qc-1.png" alt="Distribution of various QC metrics in the PBMC dataset after cell calling. Each point is a cell and is colored according to whether it was discarded by the mitochondrial filter." width="1000"><p class="caption">
Distribution of various QC metrics in the PBMC dataset after cell
calling. Each point is a cell and is colored according to whether it was
discarded by the mitochondrial filter.
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">unfiltered</span>, x<span class="op">=</span><span class="st">"sum"</span>, y<span class="op">=</span><span class="st">"subsets_Mito_percent"</span>,</span>
<span>    colour_by<span class="op">=</span><span class="st">"discard"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="tenx-unfiltered-pbmc4k_files/figure-html/unref-unfiltered-pbmc-mito-1.png" alt="Proportion of mitochondrial reads in each cell of the PBMC dataset compared to its total count." width="1000"><p class="caption">
Proportion of mitochondrial reads in each cell of the PBMC dataset
compared to its total count.
</p>
</div>
</div>
<div class="section level2">
<h2 id="normalization">Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/quickCluster.html" class="external-link">quickCluster</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/computeSumFactors.html" class="external-link">computeSumFactors</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, cluster<span class="op">=</span><span class="va">clusters</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">sizeFactors</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </span></span>
<span><span class="co">##  0.00749  0.71207  0.87490  1.00000  1.09900 12.25412</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/librarySizeFactors.html" class="external-link">librarySizeFactors</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">sizeFactors</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span>, pch<span class="op">=</span><span class="fl">16</span>,</span>
<span>    xlab<span class="op">=</span><span class="st">"Library size factors"</span>, ylab<span class="op">=</span><span class="st">"Deconvolution factors"</span>, log<span class="op">=</span><span class="st">"xy"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="tenx-unfiltered-pbmc4k_files/figure-html/unref-unfiltered-pbmc-norm-1.png" alt="Relationship between the library size factors and the deconvolution size factors in the PBMC dataset." width="1000"><p class="caption">
Relationship between the library size factors and the deconvolution size
factors in the PBMC dataset.
</p>
</div>
</div>
<div class="section level2">
<h2 id="variance-modelling">Variance modelling<a class="anchor" aria-label="anchor" href="#variance-modelling"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1001</span><span class="op">)</span></span>
<span><span class="va">dec.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVarByPoisson.html" class="external-link">modelGeneVarByPoisson</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span></span>
<span><span class="va">top.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html" class="external-link">getTopHVGs</a></span><span class="op">(</span><span class="va">dec.pbmc</span>, prop<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">dec.pbmc</span><span class="op">$</span><span class="va">mean</span>, <span class="va">dec.pbmc</span><span class="op">$</span><span class="va">total</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>    xlab<span class="op">=</span><span class="st">"Mean of log-expression"</span>, ylab<span class="op">=</span><span class="st">"Variance of log-expression"</span><span class="op">)</span></span>
<span><span class="va">curfit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">dec.pbmc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/curve.html" class="external-link">curve</a></span><span class="op">(</span><span class="va">curfit</span><span class="op">$</span><span class="fu">trend</span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, col<span class="op">=</span><span class="st">'dodgerblue'</span>, add<span class="op">=</span><span class="cn">TRUE</span>, lwd<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="tenx-unfiltered-pbmc4k_files/figure-html/unref-unfiltered-pbmc-var-1.png" alt="Per-gene variance as a function of the mean for the log-expression values in the PBMC dataset. Each point represents a gene (black) with the mean-variance trend (blue) fitted to simulated Poisson counts." width="1000"><p class="caption">
Per-gene variance as a function of the mean for the log-expression
values in the PBMC dataset. Each point represents a gene (black) with
the mean-variance trend (blue) fitted to simulated Poisson counts.
</p>
</div>
</div>
<div class="section level2">
<h2 id="dimensionality-reduction">Dimensionality reduction<a class="anchor" aria-label="anchor" href="#dimensionality-reduction"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">10000</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/denoisePCA.html" class="external-link">denoisePCA</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, subset.row<span class="op">=</span><span class="va">top.pbmc</span>, technical<span class="op">=</span><span class="va">dec.pbmc</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">100000</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runTSNE.html" class="external-link">runTSNE</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, dimred<span class="op">=</span><span class="st">"PCA"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1000000</span><span class="op">)</span></span>
<span><span class="va">sce.pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html" class="external-link">runUMAP</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, dimred<span class="op">=</span><span class="st">"PCA"</span><span class="op">)</span></span></code></pre></div>
<p>We verify that a reasonable number of PCs is retained.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, <span class="st">"PCA"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 9</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="clustering">Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html" class="external-link">buildSNNGraph</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, k<span class="op">=</span><span class="fl">10</span>, use.dimred <span class="op">=</span> <span class="st">'PCA'</span><span class="op">)</span></span>
<span><span class="va">clust</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_walktrap.html" class="external-link">cluster_walktrap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span><span class="op">$</span><span class="va">membership</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html" class="external-link">colLabels</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">clust</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html" class="external-link">colLabels</a></span><span class="op">(</span><span class="va">sce.pbmc</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16 </span></span>
<span><span class="co">## 205 508 541  56 374 125  46 432 302 867  47 155 166  61  84  16</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html" class="external-link">plotTSNE</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, colour_by<span class="op">=</span><span class="st">"label"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="tenx-unfiltered-pbmc4k_files/figure-html/unref-unfiltered-pbmc-tsne-1.png" alt="Obligatory $t$-SNE plot of the PBMC dataset, where each point represents a cell and is colored according to the assigned cluster." width="1000"><p class="caption">
Obligatory <span class="math inline">\(t\)</span>-SNE plot of the PBMC
dataset, where each point represents a cell and is colored according to
the assigned cluster.
</p>
</div>
</div>
<div class="section level2">
<h2 id="interpretation">Interpretation<a class="anchor" aria-label="anchor" href="#interpretation"></a>
</h2>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/findMarkers.html" class="external-link">findMarkers</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, pval.type<span class="op">=</span><span class="st">"some"</span>, direction<span class="op">=</span><span class="st">"up"</span><span class="op">)</span></span></code></pre></div>
<p>We examine the markers for cluster 8 in more detail. High expression
of <em>CD14</em>, <em>CD68</em> and <em>MNDA</em> combined with low
expression of <em>CD16</em> suggests that this cluster contains
monocytes, compared to macrophages in cluster 15 (Figure
@ref(fig:unref-mono-pbmc-markers)).</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">marker.set</span> <span class="op">&lt;-</span> <span class="va">markers</span><span class="op">[[</span><span class="st">"8"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">marker.set</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                     p.value           FDR summary.logFC</span></span>
<span><span class="co">## CSTA          7.170624e-222 2.015964e-217     2.4178954</span></span>
<span><span class="co">## MNDA          1.196631e-221 2.015964e-217     2.6614935</span></span>
<span><span class="co">## FCN1          2.375980e-213 2.668543e-209     2.6380934</span></span>
<span><span class="co">## S100A12       4.393470e-212 3.700839e-208     3.0808902</span></span>
<span><span class="co">## VCAN          1.711043e-199 1.153038e-195     2.2603760</span></span>
<span><span class="co">## TYMP          1.173532e-154 6.590164e-151     2.0237930</span></span>
<span><span class="co">## AIF1          3.673649e-149 1.768285e-145     2.4603604</span></span>
<span><span class="co">## LGALS2        4.004740e-137 1.686696e-133     1.8927606</span></span>
<span><span class="co">## MS4A6A        5.639909e-134 2.111457e-130     1.5457061</span></span>
<span><span class="co">## FGL2          2.044513e-124 6.888781e-121     1.3859366</span></span>
<span><span class="co">## RP11-1143G9.4 6.891551e-122 2.110945e-118     2.8042347</span></span>
<span><span class="co">## AP1S2         1.786019e-112 5.014842e-109     1.7703547</span></span>
<span><span class="co">## CD14          1.195352e-110 3.098169e-107     1.4259764</span></span>
<span><span class="co">## CFD           6.870490e-109 1.653531e-105     1.3560255</span></span>
<span><span class="co">## GPX1          9.048825e-107 2.032607e-103     2.4013937</span></span>
<span><span class="co">## TNFSF13B       3.920319e-95  8.255701e-92     1.1151275</span></span>
<span><span class="co">## KLF4           3.309726e-94  6.559876e-91     1.2049050</span></span>
<span><span class="co">## GRN            4.801206e-91  8.987324e-88     1.3814668</span></span>
<span><span class="co">## NAMPT          2.489624e-90  4.415020e-87     1.1438687</span></span>
<span><span class="co">## CLEC7A         7.736088e-88  1.303299e-84     1.0616120</span></span>
<span><span class="co">## S100A8         3.124930e-84  5.013875e-81     4.8051993</span></span>
<span><span class="co">## SERPINA1       1.580359e-82  2.420392e-79     1.3842689</span></span>
<span><span class="co">## CD36           8.018347e-79  1.174653e-75     1.0538169</span></span>
<span><span class="co">## MPEG1          8.481588e-79  1.190744e-75     0.9778095</span></span>
<span><span class="co">## CD68           5.118714e-78  6.898798e-75     0.9481203</span></span>
<span><span class="co">## CYBB           1.200516e-77  1.555776e-74     1.0300245</span></span>
<span><span class="co">## S100A11        1.174556e-72  1.465759e-69     1.8962486</span></span>
<span><span class="co">## RBP7           2.467027e-71  2.968714e-68     0.9666127</span></span>
<span><span class="co">## BLVRB          3.762610e-71  4.371634e-68     0.9701168</span></span>
<span><span class="co">## CD302          9.859086e-71  1.107307e-67     0.8792077</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotExpression.html" class="external-link">plotExpression</a></span><span class="op">(</span><span class="va">sce.pbmc</span>, features<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CD14"</span>, <span class="st">"CD68"</span>,</span>
<span>    <span class="st">"MNDA"</span>, <span class="st">"FCGR3A"</span><span class="op">)</span>, x<span class="op">=</span><span class="st">"label"</span>, colour_by<span class="op">=</span><span class="st">"label"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="tenx-unfiltered-pbmc4k_files/figure-html/unref-mono-pbmc-markers-1.png" alt="Distribution of expression values for monocyte and macrophage markers across clusters in the PBMC dataset." width="1000"><p class="caption">
Distribution of expression values for monocyte and macrophage markers
across clusters in the PBMC dataset.
</p>
</div>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>R version 4.2.2 (2022-10-31) Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.1 LTS</p>
<p>Matrix products: default BLAS:
/usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 LAPACK:
/usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</p>
<p>locale: [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C<br>
[3] LC_TIME=en_US.UTF-8 LC_COLLATE=en_US.UTF-8<br>
[5] LC_MONETARY=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8<br>
[7] LC_PAPER=en_US.UTF-8 LC_NAME=C<br>
[9] LC_ADDRESS=C LC_TELEPHONE=C<br>
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C</p>
<p>attached base packages: [1] stats4 stats graphics grDevices utils
datasets methods<br>
[8] base</p>
<p>other attached packages: [1] scran_1.26.0
EnsDb.Hsapiens.v86_2.99.0<br>
[3] ensembldb_2.22.0 AnnotationFilter_1.22.0<br>
[5] GenomicFeatures_1.50.2 AnnotationDbi_1.60.0<br>
[7] scater_1.26.1 ggplot2_3.4.0<br>
[9] scuttle_1.8.1 DropletUtils_1.18.1<br>
[11] SingleCellExperiment_1.20.0 SummarizedExperiment_1.28.0 [13]
Biobase_2.58.0 GenomicRanges_1.50.1<br>
[15] GenomeInfoDb_1.34.4 IRanges_2.32.0<br>
[17] S4Vectors_0.36.0 BiocGenerics_0.44.0<br>
[19] MatrixGenerics_1.10.0 matrixStats_0.63.0<br>
[21] DropletTestFiles_1.8.0 knitr_1.41</p>
<p>loaded via a namespace (and not attached): [1] AnnotationHub_3.6.0
BiocFileCache_2.6.0<br>
[3] systemfonts_1.0.4 igraph_1.3.5<br>
[5] lazyeval_0.2.2 BiocParallel_1.32.4<br>
[7] digest_0.6.30 htmltools_0.5.3<br>
[9] viridis_0.6.2 fansi_1.0.3<br>
[11] magrittr_2.0.3 memoise_2.0.1<br>
[13] ScaledMatrix_1.6.0 cluster_2.1.4<br>
[15] limma_3.54.0 Biostrings_2.66.0<br>
[17] R.utils_2.12.2 pkgdown_2.0.6<br>
[19] prettyunits_1.1.1 colorspace_2.0-3<br>
[21] blob_1.2.3 rappdirs_0.3.3<br>
[23] ggrepel_0.9.2 textshaping_0.3.6<br>
[25] xfun_0.35 dplyr_1.0.10<br>
[27] crayon_1.5.2 RCurl_1.98-1.9<br>
[29] jsonlite_1.8.3 glue_1.6.2<br>
[31] gtable_0.3.1 zlibbioc_1.44.0<br>
[33] XVector_0.38.0 DelayedArray_0.24.0<br>
[35] BiocSingular_1.14.0 Rhdf5lib_1.20.0<br>
[37] HDF5Array_1.26.0 scales_1.2.1<br>
[39] DBI_1.1.3 edgeR_3.40.0<br>
[41] Rcpp_1.0.9 viridisLite_0.4.1<br>
[43] xtable_1.8-4 progress_1.2.2<br>
[45] dqrng_0.3.0 bit_4.0.5<br>
[47] rsvd_1.0.5 metapod_1.6.0<br>
[49] httr_1.4.4 FNN_1.1.3.1<br>
[51] ellipsis_0.3.2 farver_2.1.1<br>
[53] pkgconfig_2.0.3 XML_3.99-0.12<br>
[55] R.methodsS3_1.8.2 uwot_0.1.14<br>
[57] sass_0.4.4 dbplyr_2.2.1<br>
[59] locfit_1.5-9.6 utf8_1.2.2<br>
[61] labeling_0.4.2 tidyselect_1.2.0<br>
[63] rlang_1.0.6 later_1.3.0<br>
[65] munsell_0.5.0 BiocVersion_3.16.0<br>
[67] tools_4.2.2 cachem_1.0.6<br>
[69] cli_3.4.1 generics_0.1.3<br>
[71] RSQLite_2.2.19 ExperimentHub_2.6.0<br>
[73] evaluate_0.18 stringr_1.4.1<br>
[75] fastmap_1.1.0 yaml_2.3.6<br>
[77] ragg_1.2.4 bit64_4.0.5<br>
[79] fs_1.5.2 purrr_0.3.5<br>
[81] KEGGREST_1.38.0 sparseMatrixStats_1.10.0<br>
[83] mime_0.12 R.oo_1.25.0<br>
[85] xml2_1.3.3 biomaRt_2.54.0<br>
[87] compiler_4.2.2 beeswarm_0.4.0<br>
[89] filelock_1.0.2 curl_4.3.3<br>
[91] png_0.1-8 interactiveDisplayBase_1.36.0 [93] statmod_1.4.37
tibble_3.1.8<br>
[95] bslib_0.4.1 stringi_1.7.8<br>
[97] highr_0.9 desc_1.4.2<br>
[99] bluster_1.8.0 lattice_0.20-45<br>
[101] ProtGenerics_1.30.0 Matrix_1.5-3<br>
[103] vctrs_0.5.1 pillar_1.8.1<br>
[105] lifecycle_1.0.3 rhdf5filters_1.10.0<br>
[107] BiocManager_1.30.19 jquerylib_0.1.4<br>
[109] BiocNeighbors_1.16.0 bitops_1.0-7<br>
[111] irlba_2.3.5.1 httpuv_1.6.6<br>
[113] rtracklayer_1.58.0 R6_2.5.1<br>
[115] BiocIO_1.8.0 promises_1.2.0.1<br>
[117] gridExtra_2.3 vipor_0.4.5<br>
[119] codetools_0.2-18 assertthat_0.2.1<br>
[121] rhdf5_2.42.0 rprojroot_2.0.3<br>
[123] rjson_0.2.21 withr_2.5.0<br>
[125] GenomicAlignments_1.34.0 Rsamtools_2.14.0<br>
[127] GenomeInfoDbData_1.2.9 parallel_4.2.2<br>
[129] hms_1.1.2 grid_4.2.2<br>
[131] beachmat_2.14.0 rmarkdown_2.18<br>
[133] DelayedMatrixStats_1.20.0 Rtsne_0.16<br>
[135] shiny_1.7.3 ggbeeswarm_0.6.0<br>
[137] restfulr_0.0.15</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Aedin Culhane, Michael Lynch.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
